diff --git a/cpp/open3d/visualization/rendering/Renderer.h b/cpp/open3d/visualization/rendering/Renderer.h
index fc7876621..597636c80 100644
--- a/cpp/open3d/visualization/rendering/Renderer.h
+++ b/cpp/open3d/visualization/rendering/Renderer.h
@@ -7,6 +7,7 @@
 
 #pragma once
 
+#include <memory>
 #include "open3d/visualization/rendering/MaterialModifier.h"
 #include "open3d/visualization/rendering/RendererHandle.h"
 
