diff --git a/3rdparty/find_dependencies.cmake b/3rdparty/find_dependencies.cmake
index 169bfbe1a..e4b7cafb8 100644
--- a/3rdparty/find_dependencies.cmake
+++ b/3rdparty/find_dependencies.cmake
@@ -1135,6 +1135,8 @@ if(WITH_MINIZIP)
     )
     list(APPEND Open3D_3RDPARTY_PRIVATE_TARGETS_FROM_SYSTEM Open3D::3rdparty_minizip)
 endif()
+find_package(minizip CONFIG REQUIRED)
+list(APPEND Open3D_3RDPARTY_PRIVATE_TARGETS_FROM_SYSTEM MINIZIP::minizip)
 
 # Googletest
 if (BUILD_UNIT_TESTS)
